src/Features/Core/Portable/Completion/Providers/TypeImportCompletionService.cs

```diff
@@ -0,0 +1,349 @@
+                }
+            }
+
+            private static void GetAccessibleTopLevelTypesWorker<TKey>(
+                TKey key,
+                INamespaceSymbol rootNamespace,
+                Checksum checksum,
+                bool isInternalsVisible,
+                Action<CompletionItem> handleAccessibleItem,
+                ConcurrentDictionary<TKey, ReferenceCacheEntry> cache,
+                CancellationToken cancellationToken)
+            {
+                // Cache miss, create all requested items.
+                if (!cache.TryGetValue(key, out var cacheEntry) ||
+                    cacheEntry.Checksum != checksum)
+                {
+                    var items = GetCompletionItemsForTopLevelTypeDeclarations(rootNamespace, cancellationToken);
+                    cacheEntry = new ReferenceCacheEntry(checksum, items);
+                    cache[key] = cacheEntry;
+                }
+
+                HandleAccessibleItems(cacheEntry.CachedItems, isInternalsVisible, handleAccessibleItem);
+            }
+
+            private static ImmutableArray<CompletionItemEntry> GetCompletionItemsForTopLevelTypeDeclarations(
+                INamespaceSymbol rootNamespaceSymbol,
+                CancellationToken cancellationToken)
+            {
+                var builder = ArrayBuilder<CompletionItemEntry>.GetInstance();
+                VisitNamespace(rootNamespaceSymbol, containingNamespace: null, builder, cancellationToken);
+                return builder.ToImmutableAndFree();
+
+                static void VisitNamespace(
+                    INamespaceSymbol symbol,
+                    string containingNamespace,
+                    ArrayBuilder<CompletionItemEntry> builder,
+                    CancellationToken cancellationToken)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    containingNamespace = ConcatNamespace(containingNamespace, symbol.Name);
+
+                    foreach (var memberNamespace in symbol.GetNamespaceMembers())
+                    {
+                        VisitNamespace(memberNamespace, containingNamespace, builder, cancellationToken);
+                    }
+
+                    var overloads = PooledDictionary<string, TypeOverloadInfo>.GetInstance();
+                    var memberTypes = symbol.GetTypeMembers();
+
+                    // Iterate over all top level internal and public types, keep track of "type overloads".
+                    foreach (var memberType in memberTypes)
+                    {
+                        if (IsAccessible(memberType.DeclaredAccessibility) && memberType.CanBeReferencedByName)
+                        {
+                            overloads.TryGetValue(memberType.Name, out var overloadInfo);
+                            overloads[memberType.Name] = overloadInfo.Aggregate(memberType);
+                        }
+                    }
+
+                    foreach (var pair in overloads)
```

Comment:
consider `foreach (var (_, overloadInfo) in overloads)`

Reply:
it complains about no Deconstruct instance for KeyValuePair

Reply:
Import `Roslyn.Utilities`.